### PR TITLE
Feature/request lib

### PR DESCRIPTION
**Pull request by tnagorra**

### CODE REVIEW

File: src/components/other/ExportPreview/index.tsx [hunk 1]
```diff
@@ -49,7 +49,10 @@ function ExportPreview(props: OwnProps) {
         onSuccess: (response) => {
             setExportObj(response);
         },
-        onFailure: () => {
+        onFailure: (test) => {
+            console.log(
+                test.value.faramErrors,
+            );
```
Comment (samshara) [inline src/components/other/ExportPreview/index.tsx]: remove console.log

File: package.json [hunk 2]
```diff
@@ -13,9 +13,9 @@
     "scripts": {
         "start": "webpack-dev-server --mode development --config config/webpack.dev.js --env.NODE_ENV=development --history-api-fallback",
         "build": "webpack --mode production --config config/webpack.prod.js --env.NODE_ENV=production",
-        "typecheck": "tsc",
+        "typecheck": "tsc --incremental",
```
Comment (AdityaKhatri) [inline package.json:16]: What does incremental do?

File: src/App.js [hunk 3]
```diff
@@ -161,9 +166,12 @@ export default class App extends React.PureComponent {
             return <AppLoading />;
         }
 
+        // FIXME: memoize this
```
Comment (AdityaKhatri) [inline src/App.js:169]: Please do in this PR

File: src/components/other/ExportPreview/index.tsx (hunk 1 above)
Comment (tnagorra) [reply src/components/other/ExportPreview/index.tsx]: oh no! yes sir!

File: src/utils/request/fetch.ts [hunk 4]
```diff
@@ -111,53 +103,52 @@ async function fetchResource<T, Q>(
         if (!signal.aborted) {
             console.error(`An error occurred while fetching ${myUrl}`, e);
 
-            const message = {
-                reason: 'network',
-                exception: e,
-                value: { nonFieldErrors: ['Network error'] },
-            };
-            await handleError(message);
+            const terror = transformErrorRef.current(
```
Comment (AdityaKhatri) [inline src/utils/request/fetch.ts]: what is terror?

File: src/utils/request/fetch.ts [hunk 5]
```diff
@@ -111,53 +103,52 @@ async function fetchResource<T, Q>(
...
 
     if (!res.ok) {
-        const message = {
-            reason: 'other',
-            exception: undefined,
-            value: (resBody as { errors: Err }).errors,
-            errorCode: (resBody as { errors: Err; errorCode: number }).errorCode,
-        };
-        await handleError(message);
+        const tError = transformErrorRef.current(
```
Comment (AdityaKhatri) [inline src/utils/request/fetch.ts]: @tnagorra tError?

File: package.json (hunk 2 above)
Comment (tnagorra) [reply package.json:16]: It does incremental typecheck. Please read docs sir!

File: src/utils/request/fetch.ts (hunk 4 above)
Comment (tnagorra) [reply src/utils/request/fetch.ts]: transformError

File: src/utils/request/fetch.ts (hunk 5 above)
Comment (tnagorra) [reply src/utils/request/fetch.ts]: transformError

File: src/App.js (hunk 3 above)
Comment (tnagorra) [reply src/App.js:169]: Can't do right now. We need to first convert App.js to functional component!

**Review by AdityaKhatri — APPROVED** [review]

**Pull request merged**